utils/test_results.py

```diff
@@ -59,15 +58,201 @@ class GENERATE_TEST_RESULT_PARAM:
     SKIPPED = "skipped"
 
 
+@dataclass
+class TestResultsQuery:
+    query: str
+    params: dict[str, int | str | tuple[str, ...]]
+
+
+def convert_tuple_or_none(value: set[str] | list[str] | None) -> tuple[str, ...] | None:
+    return tuple(value) if value else None
+
+
+def encode_after_or_before(value: str | None) -> str | None:
```

Comment:
this function is actually just encode_to_b64 isnt it? Nothing in particular to after or before